### ISSUE TITLE
[Enhancement] Rules - Card contains {value}

### ISSUE THREAD

**Issue by sfahrenholz** [wekan/wekan#2194]
**Problem description**:

i want to add some labels to a new card on lane _Backlog_. Now i want to create a rule thats make the steps: 
if a card add to lane _Backlog_ AND contains value _Patch_ so add the label _Patch_ and _Bug_ 

- currently there is no "contains value"-rule possible

**Comment by sfahrenholz** [comment]
It would be also very nice, if there are multiple values with AND or OR possible
* "Patch" AND "important"
* "Bug" OR "Fix"

**Comment by Majed6** [comment]
@xet7 Hi,

Thinking out load here. I'm interested in implementing regex support for all Rule filters. I would like to hear your opinion on:
1- Updating `buildMatchingFieldsMap` to have it generate a ``{$where:   "'activity.X'.match(`^${this.X}$`) || 'activity.Y'.match(`^${this.Y}$`)"}``.
2- A migration that escapes all filterable triggers' attributes for backward consistency. 
2- A migration that converts the escaped `\*` to `.*` for all filterable triggers' attributes for backward compatibility.
4- Switching the default filter `|| '*'` to `|| '.*'` .
5- Relying on mongodb's  11601 "JavaScript execution interrupted" exception or using `.maxTimeMS` to contain ReDoS.

Trello's butler allows for quantifiers and anchors via menus instead of straight up regex. The way I see those menus is that's just re-inventing Regex and a potential time sink.

**Comment by xet7** [comment]
@Majed6 

Thanks! Yes, you can send pull request.

**Comment by hradec** [comment]
This actually got me wondering: Does a rule accept the name of a column as regex at all? 

for example: 
![image](https://user-images.githubusercontent.com/806743/122303661-be7b7b80-ceb8-11eb-8834-bef7a3d5d86f.png)

So in this example, I want to create a rule to remove all labels from any card when it is moved to ANY column that has the word "BKP" in it's name. 

I've tried the `.*BKP.*` regex, as you can see in the example, and it doesn't seem to work. 

@Majed6 : would your change allow for this example to work too?

**Comment by Majed6** [comment]
@hradec Yes, the proposed solution should allow it to work.. It saddens me to tell you that I have moved on to building an alternative UI to the website I use. Having the two websites sync'ed just to get the beauty of Wekan involved a bit of over-engineering on my part.

The steps to make regex work on Wekan are explained above and should in theory work just fine when implemented. 
The premise of the implementation is to store those regexps  in the database and run the query `{$where: "'activity.X'.match(`^${this.X}$`) || 'activity.Y'.match(`^${this.Y}$`)"}` against them . If you're interested have a look at Wekan source code and try to understand why it's not working right now and figure out a solution to the problem. If none comes to mind you could try mine.
